content/tutorials/use-specialized-payment-types/create-custom-transactors.md

```diff
@@ -0,0 +1,379 @@
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfOwnerNode, *page);
+    }
+    // If we succeeded, the new entry counts against the creator's reserve.
+    adjustOwnerCount(view(), sle, 1, viewJ);
+    return tesSUCCESS;
+}
+```
+
+
+## Additional Functions
+
+
+### `calculateBaseFee`
+
+The `calculateBaseFee` function calculates the XRP fee to run a transaction. The fee is usually inherited from the base transactor, but transactors such as `EscrowFinish` require additional fees.
+
+```c++
+XRPAmount
+EscrowFinish::calculateBaseFee(ReadView const& view, STTx const& tx)
+{
+    XRPAmount extraFee{0};
+
+    if (auto const fb = tx[~sfFulfillment])
+    {
+        extraFee += view.fees().base * (32 + (fb->size() / 16));
+    }
+
+    return Transactor::calculateBaseFee(view, tx) + extraFee;
+}
+```
+
+
+### `makeTxConsequences`
+
+`rippled` uses a [`TxConsequences`](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/applySteps.h#L41-L44) class to describe the outcome to an account when applying a transaction. It tracks the fee, maximum possible XRP spent, and how many sequences are consumed by the transaction. There are three types of consequences:
+
+- **Normal:** The transactor doesn't affect transaction signing and only consumes an XRP fee.
+- **Blocker:** The transactor affects transaction signing, preventing valid transactions from queueing behind it.
+- **Custom:** The transactor needs to do additional work to determination consequences.
+
+The `makeTxConsequences` function enables you to create custom consequences for situations such as:
+
+- Payments sending XRP.
+- Tickets consuming more than one sequence.
+- Transactions that are normal or blockers, depending on flags or fields set.
+
+```c++
+SetAccount::makeTxConsequences(PreflightContext const& ctx)
+{
+    // The SetAccount may be a blocker, but only if it sets or clears
+    // specific account flags.
+    auto getTxConsequencesCategory = [](STTx const& tx) {
+        if (std::uint32_t const uTxFlags = tx.getFlags();
+            uTxFlags & (tfRequireAuth | tfOptionalAuth))
+            return TxConsequences::blocker;
+
+        if (auto const uSetFlag = tx[~sfSetFlag]; uSetFlag &&
+            (*uSetFlag == asfRequireAuth || *uSetFlag == asfDisableMaster ||
+             *uSetFlag == asfAccountTxnID))
+            return TxConsequences::blocker;
+
+        if (auto const uClearFlag = tx[~sfClearFlag]; uClearFlag &&
+            (*uClearFlag == asfRequireAuth || *uClearFlag == asfDisableMaster ||
+             *uClearFlag == asfAccountTxnID))
+            return TxConsequences::blocker;
+
+        return TxConsequences::normal;
+    };
+
+    return TxConsequences{ctx.tx, getTxConsequencesCategory(ctx.tx)};
+}
+```
+
+
+## Next Steps
+
+Submit your transactor to the XRPL `Devnet` for others to test. See: [Contribute Code to the XRP Ledger](contribute-code-flow.html).
```

Comment:
It's nonsensical to say, "submit the transactor to the XRPL's `Devnet`". You can't just... submit raw code to a network, let alone Devnet which is a public test network Ripple runs using specifically code that's been merged to the `develop` branch of the XRPLF `rippled` repository.

I think a more reasonable set of next steps include re-compiling the server and testing the transaction in stand-alone mode. If you put the transaction behind an amendment, you'll need to [force-enable the feature using the config file](https://xrpl.org/test-amendments.html#test-amendments).